fcsparser/api.py

```diff
@@ -27,11 +27,39 @@
 
 def fromfile(file, dtype, count, *args, **kwargs):
     """Wrapper around np.fromfile to support any file-like object."""
+    
+    dtypes = dtype.split(',')
+    field_width = []
+    
+    for dt in dtypes:
+        num_bytes = int(dt[2:])
+        field_width.append(num_bytes)
+        
     try:
-        return numpy.fromfile(file, dtype=dtype, count=count, *args, **kwargs)
+        ret = numpy.fromfile(file, 
+                             dtype=",".join(['u1'] * sum(field_width)), 
+                             count=count, 
+                             *args, 
+                             **kwargs)
     except (TypeError, IOError):
-        return numpy.frombuffer(file.read(count * numpy.dtype(dtype).itemsize),
-                                dtype=dtype, count=count, *args, **kwargs)
+        ret = numpy.frombuffer(file.read(count * sum(field_width)),
+                               dtype=",".join(['u1'] * sum(field_width)), 
+                               count=count, 
+                               *args, 
+                               **kwargs)
+
+    ret = ret.view('u1').reshape((count, sum(field_width)))
+    ret_dtypes = []
+    for field, dt in enumerate(dtypes):
+        dtype_type = dt[1]
+        dtype_endian = dt[0]
+        num_bytes = int(dt[2:])
+        while num_bytes & (num_bytes - 1) != 0:
+            ret = np.insert(ret, sum(field_width[0:field]), np.zeros(count), axis = 1)
```

Comment:
nit: https://www.python.org/dev/peps/pep-0008/#other-recommendations

axis=1 (rather than axis = 1) to follow PEP-8 guidelines